src/GafferCycles/IECoreCyclesPreview/SocketAlgo.cpp

```diff
@@ -525,6 +535,120 @@ void setRampSocket( ccl::Node *node, const ccl::SocketType *socket, const IECore
 	node->set( *socket, ramp );
 }
 
+ccl::ParamValue setParamValue( const IECore::InternedString &name, const IECore::Data *value )
+{
+	switch( value->typeId() )
+	{
+		case BoolDataTypeId :
+			if( const BoolData *data = static_cast<const BoolData *>( value ) )
```

Comment:
`static_cast()` doesn't do any runtime checks, so the only way it can return `nullptr` is if it is given `nullptr`, which is not the case here (we've already dereferenced `value` at L540). So we should drop the `if()` from each of these cases.